dev_docs/tutorials/elasticsearch.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: kibDevTutorialElasticsearchClient
+slug: /kibana-dev-docs/tutorials/elasticsearch-client
+title: Access the Elasticsearch Client in a Plugin
+description: Learn how to use the Elasticsearch client in your Kibana plugin.
+date: 2025-06-19
+tags: ['kibana','onboarding', 'dev', 'elasticsearch', 'tutorials']
+---
+
+This guide covers how to obtain the Elasticsearch client in both your plugin's start lifecycle and route handler context. It explains the difference between asCurrentUser and asInternalUser, and demonstrates best practices for robust concurrent index initialization.
+
+## Getting the Elasticsearch Client
+
+### From the Plugin Start Contract
+
+In your plugin's `start` method, the `core` parameter provides access to the Elasticsearch client:
+
+```ts
+import type { CoreStart, Plugin } from '@kbn/core/server';
+
+export class MyPlugin implements Plugin {
+  public start(core: CoreStart) {
+    const esClient = core.elasticsearch.client.asCurrentUser;
```

Comment:
```suggestion
    const esClient = core.elasticsearch.client.asInternalUser;
```

Are you thinking of `asInternalUser` since there is no request/current user in this context?